src/vm/core.zig

```diff
@@ -37,17 +41,23 @@ pub const CairoVM = struct {
     /// - `CairoVM`: The created VM.
     /// # Errors
     /// - If a memory allocation fails.
-    pub fn init(allocator: *const Allocator) !CairoVM {
+    pub fn init(allocator: *const Allocator, config: Config) !CairoVM {
         // Initialize the memory segment manager.
         const memory_segment_manager = try segments.MemorySegmentManager.init(allocator);
         // Initialize the run context.
         const run_context = try RunContext.init(allocator);
 
+        var trace: ?ArrayList(TraceEntry) = null;
+        if (config.enable_trace) {
+            trace = try ArrayList(TraceEntry).initCapacity(allocator.*, 4096);
```

Comment:
can you write a TODO comment with a link to the issue to make the initial capacity configurable please ?